src/common/snippets/tests/src/lowered/pass/move_scalar_to_consumer.cpp

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "lir_test_utils.hpp"
+
+#include "openvino/opsets/opset10.hpp"
+#include "snippets/lowered/pass/move_scalar_to_consumer.hpp"
+#include "snippets/op/load.hpp"
+#include "snippets/op/store.hpp"
+#include "snippets/op/scalar.hpp"
+
+namespace ov {
+namespace test {
+namespace snippets {
+
+using namespace ov::snippets::lowered;
+using namespace ov::snippets::lowered::pass;
+
+TEST_F(LoweredPassTestsF, MoveScalarToConsumer) {
+    const auto input_precision = ov::element::i8;
+    const ov::Shape scalar_shape{1};
+    {
+        auto scalar = linear_ir->push_node<ov::snippets::op::Scalar>(input_precision, scalar_shape, 42.f);
+        auto add1 = linear_ir->push_node<ov::opset10::Add>(scalar.second, scalar.second);
+        auto add2 = linear_ir->push_node<ov::opset10::Add>(add1.second, scalar.second);
+        auto result = linear_ir->push_node<ov::opset10::Result>(add2.second);
+    }
+    pipeline.register_pass<MoveScalarToConsumer>();
+    {
+        // No changes in IR are expected
+        auto scalar = linear_ir_ref->push_node<ov::snippets::op::Scalar>(input_precision, scalar_shape, 42.f);
+        auto add1 = linear_ir_ref->push_node<ov::opset10::Add>(scalar.second, scalar.second);
+        auto add2 = linear_ir_ref->push_node<ov::opset10::Add>(add1.second, scalar.second);
+        auto result = linear_ir_ref->push_node<ov::opset10::Result>(add2.second);
+    }
```

Comment:
If nothing should be happened (reference LIR should be as initial LIR), you can miss reference initialization- it will be done [automatically ](https://github.com/openvinotoolkit/openvino/blob/2453a57170b3530e4ceb7938848c064eda9c1c85/src/common/snippets/tests/src/lir_test_utils.cpp#L33)

Reply:
Or I suggest to add `Relu` before first `Add` and after target `Scalar`. Then the pass should move `Scalar` after `Relu` - the reference LIR should have such expression order (when the pass really did something)

Reply:
Added `Relu` and made order verification